core/src/main/java/com/linecorp/armeria/client/retry/AbstractRetryingClient.java

```diff
@@ -122,6 +119,14 @@ protected final RetryRule retryRule() {
         return retryRule;
     }
 
+    /**
+     * Fetches the {@link RetryConfig} that was mapped by the configured {@link RetryConfigMapping} for a given
+     * logical request.
+     */
+    protected final RetryConfig<O> mappedRetryConfig(ClientRequestContext ctx) {
```

Comment:
I prefer to make this internal use only since a NullPointerException will be used if this method is invoked before `State` is set.
```suggestion
    final RetryConfig<O> mappedRetryConfig(ClientRequestContext ctx) {
```

Reply:
Let's do it.